src/deck/backends/deck_backend_deckctrl.c

```diff
@@ -209,27 +214,37 @@ static DeckInfo* deckctrl_getNextDeck(void)
     }
 
     // Print ID and new I2C address of the detected deck if debugging is enabled
-    DEBUG_PRINT("Found deck with ID: %02x %02x %02x %02x %02x %02x %02x %02x %02x %02x %02x %02x\n",
+    DECKCTRL_DEBUG_PRINT("Found deck with ID: %02x %02x %02x %02x %02x %02x %02x %02x %02x %02x %02x %02x\n",
                 cpu_id[0], cpu_id[1], cpu_id[2], cpu_id[3],
                 cpu_id[4], cpu_id[5], cpu_id[6], cpu_id[7],
                 cpu_id[8], cpu_id[9], cpu_id[10], cpu_id[11]);
-    DEBUG_PRINT("Assigned new I2C address: 0x%02x\n", deck_address);
+    DECKCTRL_DEBUG_PRINT("Assigned new I2C address: 0x%02x\n", deck_address);
 
     // Read deck information from the newly assigned address at register 0x0000
-    // Format: magic(2) + versions(2) + vendor(1) + product(1) + revision(1) + name(14)
-    uint8_t deck_mem_data[21];
+    // Format: magic(2) + versions(2) + vendor(1) + product(1) + revision(1) + name(14) + production date(3) + checksum(1)
+    uint8_t deck_mem_data[0x20];
```

Comment:
Summing the numbers in the comment the length should be 25 (0x19)

Reply:
Updated the comment to reflect that the config block is actually 32 bytes.